File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.java
class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        int sum=Arrays.stream(apple).sum();
        capacity=Arrays.stream(capacity).boxed().sorted(Collections.reverseOrder()).mapToInt(Integer::intValue).toArray();
        int box=0;
        for(int i=0;i<capacity.length;i++){
            sum-=capacity[i];
            box++;
            if(sum<=0) break;
        }
        return box;
    }
}

File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.py
class Solution(object):
    def minimumBoxes(self, apple, capacity):
        """
        :type apple: List[int]
        :type capacity: List[int]
        :rtype: int
        """
        sm=sum(apple)
        capacity.sort(reverse=True)
        c=0
        for i in capacity:
            sm-=i
            c+=1
            if sm<=0: break
        return c